modules/debts_api/lib/debts_api/v0/fsr_form_transform/income_calculator.rb

```diff
@@ -0,0 +1,207 @@
+
+          soc_sec_amt = if enhanced_fsr_active
+                          0
+                        elsif beneficiary_type == 'spouse'
+                          social_security.dig('spouse', 'socialSecAmt').to_f || 0
+                        else
+                          social_security['socialSecAmt'].to_f || 0
+                        end
+
+          comp = if beneficiary_type == 'spouse'
+                   benefits.dig('spouseBenefits', 'compensationAndPension').to_f || 0
+                 else
+                   income.sum { |item| item['compensationAndPension'].to_f }
+                 end
+
+          edu = if beneficiary_type == 'spouse'
+                  benefits.dig('spouseBenefits', 'education').to_f || 0
+                else
+                  income.sum { |item| item['education'].to_f }
+                end
+
+          benefits_amount = comp + edu
+
+          deductions = if enhanced_fsr_active
+                         employment_records
+                           .select { |emp| emp['isCurrent'] }
+                           .map do |emp|
+                           if emp['deductions'].nil?
+                             0
+                           else
+                             emp['deductions']
+                           end
+                         end
+                         .flatten
+                       else
+                         curr_employment.pluck('deductions').flatten
+                       end
+
+          gross_salary = gross_salary.to_f.round(2)
+          taxes_values = filter_reduce_by_name(deductions, @tax_filters)
+          retirement_values = filter_reduce_by_name(deductions, @retirement_filters)
+          social_sec = filter_reduce_by_name(deductions, @social_sec_filters)
+          other = other_deductions_amt(deductions, @all_filters)
+          tot_deductions = taxes_values + retirement_values + social_sec + other
+          other_income = addl_inc.to_f.round(2) + benefits_amount.to_f.round(2) + soc_sec_amt.to_f.round(2)
+          net_income = gross_salary - tot_deductions
+          total_monthly_net_income = net_income + other_income
+
+          {
+            grossSalary: gross_salary,
+            deductions: {
+              taxes: taxes_values,
+              retirement: retirement_values,
+              socialSecurity: social_sec,
+              otherDeductions: {
+                name: other_deductions_name(deductions, @all_filters),
+                amount: other
+              }
+            },
+            totalDeductions: tot_deductions,
+            netTakeHomePay: net_income,
+            otherIncome: {
+              name: name_str(soc_sec_amt, comp, edu, addl_inc_records),
+              amount: other_income.round(2)
+            },
+            totalMonthlyNetIncome: total_monthly_net_income.round(2)
+          }
+        end
+      end
+    end
+  end
+end
+# rubocop:enable Metrics/MethodLength
```

Comment:
Let's not disable the rubocop checks here. If you don't want to address it now you can add to the rubocop_todo.